test/csuite/wt8246_compact_rts_data_correctness/main.c

```diff
@@ -0,0 +1,375 @@
+
+    for (i = 0; i < NUM_RECORDS; i++) {
+        cursor->set_key(cursor, i + 1);
+        testutil_check(cursor->search(cursor));
+        testutil_check(cursor->get_value(cursor, &val1, &val2, &val3, &str_val));
+
+        val_2_size = strlen(str_val);
+
+        testutil_assert(val_1_size == val_2_size);
+        // printf ("str_val : %s\n", str_val);
+        testutil_assert(memcmp(value, str_val, val_1_size) == 0);
+    }
+
+    testutil_check(session->commit_transaction(session, NULL));
+    testutil_check(cursor->close(cursor));
+    cursor = NULL;
+}
+
+static void
+large_updates(WT_SESSION *session, const char *uri, char *value, int commit_ts)
+{
+    WT_CURSOR *cursor;
+    WT_RAND_STATE rnd;
+    uint64_t val;
+    int i;
+    char tscfg[64];
+
+    __wt_random_init_seed((WT_SESSION_IMPL *)session, &rnd);
+    testutil_check(session->open_cursor(session, uri, NULL, NULL, &cursor));
+
+    for (i = 0; i < NUM_RECORDS; i++) {
+        testutil_check(session->begin_transaction(session, NULL));
+        cursor->set_key(cursor, i + 1);
+        val = (uint64_t)__wt_random(&rnd);
+        cursor->set_value(cursor, val, val, val, value);
+        testutil_check(cursor->insert(cursor));
+        testutil_check(__wt_snprintf(tscfg, sizeof(tscfg), "commit_timestamp=%d", commit_ts));
+        testutil_check(session->commit_transaction(session, tscfg));
+    }
+
+    testutil_check(cursor->close(cursor));
+    cursor = NULL;
+}
+
+static void
+populate(WT_SESSION *session, const char *uri)
+{
+    WT_CURSOR *cursor;
+    WT_RAND_STATE rnd;
+    uint64_t val;
+    int i, str_len;
+
+    printf("Populating table...\n");
+
+    __wt_random_init_seed((WT_SESSION_IMPL *)session, &rnd);
+
+    str_len = sizeof(data_str) / sizeof(data_str[0]);
+    for (i = 0; i < str_len - 1; i++)
+        data_str[i] = 'a' + __wt_random(&rnd) % 26;
+
+    data_str[str_len - 1] = '\0';
+
+    testutil_check(session->open_cursor(session, uri, NULL, NULL, &cursor));
+    for (i = 0; i < NUM_RECORDS; i++) {
+        cursor->set_key(cursor, i + 1);
+        val = (uint64_t)__wt_random(&rnd);
+        cursor->set_value(cursor, val, val, val, data_str);
+        testutil_check(cursor->insert(cursor));
+    }
+
+    testutil_check(cursor->close(cursor));
+    cursor = NULL;
+}
+
+static void
+remove_records(WT_SESSION *session, const char *uri, int commit_ts)
+{
+    WT_CURSOR *cursor;
+    int i;
+    char tscfg[64];
+
+    printf("Removing records...\n");
+    testutil_check(session->open_cursor(session, uri, NULL, NULL, &cursor));
+
+    /* Remove 1/3 of the records from the middle of the key range. */
+    for (i = NUM_RECORDS / 3; i < (NUM_RECORDS * 2) / 2; i++) {
+        testutil_check(session->begin_transaction(session, NULL));
+        cursor->set_key(cursor, i);
+        testutil_check(cursor->remove(cursor));
+        testutil_check(__wt_snprintf(tscfg, sizeof(tscfg), "commit_timestamp=%d", commit_ts));
```

Comment:
Little optimization: we can move the line above outside of the loop.

Reply:
Fixed